pkg/controller/iampolicy/iampolicy_controller.go

```diff
@@ -258,7 +272,12 @@ func checkUnNamespacedPolicies(plcToUpdateMap map[string]*policiesv1.IamPolicy)
 		if policy.Spec.ClusterRole != "" {
 			clusterRoleRef = policy.Spec.ClusterRole
 		}
-		clusterLevelUsers := checkAllClusterLevel(ClusteRoleBindingList, clusterRoleRef)
+
+		clusterLevelUsers, err := checkAllClusterLevel(ClusteRoleBindingList, clusterRoleRef)
+		if err != nil {
+			log.Error(err, fmt.Sprintf("Failed to check for the users with the role %s: %v", clusterRoleRef, err))
+			continue
```

Comment:
I would think that if we had an error, we need to requeue the policy to evaluate it again. It looks like it could be a failed GET, or it could be an incorrectly formatted group object.

With the early return inside `checkAllClusterLevel`, if there is an error, we'd "find" 0 users - even if the loop in that was able to find some users without errors.

Reply:
So this doesn't run in the `Reconcile` method. It's run as a separate Go routine that runs periodically:
https://github.com/open-cluster-management/iam-policy-controller/blob/92927bf8bc2182f659751efeffb814ed8e4e89a5/pkg/controller/iampolicy/iampolicy_controller.go#L204

So my thought was to not evaluate the policy at all since we can't determine the correct amount of users at this time.

Reply:
Good point - that does make it better, since it will always be checked again anyway.

I still think we should count up as many users as we can without errors. If that number is larger than the allowed number, we can be safe in saying the policy is noncompliant. But we wouldn't want to report compliant if there are errors, because we might be under-counting.